### ISSUE TITLE
[SelectMenuOption]: AccessibleName of SelectMenuOption not announced by VoiceOver and JAWS.

### ISSUE THREAD

**Issue by witja46** [SAP/ui5-webcomponents-react#7478]
### Describe the bug

Screen readers (VoiceOver on Mac and JAWS on Windows) do not announce the accessibleName text of the SelectMenuOption in both Chrome and Firefox.

### Isolated Example

https://stackblitz.com/edit/github-d4xfy9u1?file=src%2FApp.tsx

### Reproduction steps

Using VoiceOver in Chrome:
1. Tab to the  Select component
2. Open it by pressing enter
3. Listen to the Screen Reader announcement of the first SelectMenuOption
4. Voice Over announces: "A..z list position 1 of 4"
...

### Expected Behaviour

Text specified with the accessibleName("Sort in alphabetical ascending order") is read to the user followed by the position of the option inside of the options list.

### UI5 Web Components for React Version

1.27.0

### UI5 Web Components Version

1.24.0

### Browser

Chrome

### Operating System

MacOS and Windows 11

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
